### DISCUSSION TITLE
Documentation for updating product custom fields in the new Product Block Editor?

### DISCUSSION THREAD

**Discussion by zackaira** [woocommerce/woocommerce#39140] [WooCommerce New Product Editor] [answered]
Can anyone point me in the direction with a link to updating product custom fields that have been added by my plugin?

My plugin (one feature of it) is adding pricing prefix and suffix settings to the Product Data meta box which is now not showing in the new Product Block Editor version.

I'd like to have this fixed before the update is released.

Thanks

**Comment by louwie17** [comment]
Hi @zackaira, thanks for the question and sorry for the very late reply.
Unfortunately, we don't have any specific documentation yet, so I will point you to some code examples.

In the new product block editor you can update the product data by using the `useEntityProp` hook. There is a similar example of this in the [metabox guide in Gutenberg](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/metabox.md?plain=1#L84).
You can see how we make use of that in the [price field block](https://github.com/woocommerce/woocommerce/blob/e3194c25e340e1f9d5b5d0da29eae4ec51eb726c/packages/js/product-editor/src/blocks/pricing/edit.tsx#L30-L34), where we get the `regular_price` as such:
```
const [ price, setPrice ] = useEntityProp< string >(
		'postType',
		'product',
		'regular_price'
	);
```
I am not sure if this is what you need to add a prefix/suffix? Or are you referring to the actual field it-self and not the value?
If not, a screenshot of how its expected to work in the current WP Product editor would help.

**Comment by zackaira** [comment]
Hi @louwie17 
Thanks for the reply... Sorry maybe I worded it wrong. My plugin currently adds settings into the Product Data meta box (before the new Product Block Editor), it adds a price suffix input and a price prefix input for the users to add those to the product price on the front-end.

Those setting inputs are now gone, they do not show in the new Product Block Editor.

So I'm wondering if there's updated documentation on how to add these product inputs back into the new Product Block Editor.

Hope I've made that clearer.
Thanks!

**Reply by louwie17** [reply, marked as answer]
Thanks for the clarification @zackaira. We don't have any specific documentation around inserting other fields into the form yet, as we are in the middle of simplifying how that is being done. As we speak we are putting some items in place to handle the product form template better and are migrating the product block editor to this. Once that is done we will update our documentation with some examples.

It is possible to already inject your fields into the form now, and the closest example would be this [scaffolding package](https://github.com/woocommerce/woocommerce/tree/trunk/packages/js/create-product-editor-block), it contains an example where a field is inserted in the product form: [example](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/create-product-editor-block/plugin-templates/%24slug.php.mustache#L50-L95).
But as I mentioned above, this code will change once the items I mentioned above are in place.

**Reply by zackaira** [reply]
Ok awesome... Thanks so much for letting me know @louwie17
